### PR TITLE
[BEAM-603] Update Beam examples to reflect new apply style (>>), rewording explanations, renaming variables

### PR DESCRIPTION
**Pull request by mariapython**
Updated the examples section of the readme under incubator-beam/sdks/python/
TODO (@hadarhod87):
- [x] Update Examples section of the readme 
- [ ] Update the rest of the sections.

### CODE REVIEW

**Comment by mariapython** [comment]
R: @aaltay

File: sdks/python/README.md [hunk 1]
```diff
@@ -257,40 +257,38 @@ Some interesting classes to navigate to:
...
 element of the input `PCollection` and must return an element to go
 into the output `PCollection`.
 
 ```python
-import google.cloud.dataflow as df
-p = df.Pipeline('DirectPipelineRunner')
-# Read file with names, add a greeting for each, and write results.
+import apache_beam as beam
+p = beam.Pipeline('DirectPipelineRunner')
+# Read file with names, add a greeting for each, and write to a file.
```
Comment (aaltay) [inline sdks/python/README.md]: "Read **a** file"

File: sdks/python/README.md [hunk 2]
```diff
@@ -257,40 +257,38 @@ Some interesting classes to navigate to:
 
 ## Some Simple Examples
 
-### Hello world
```
Comment (aaltay) [inline sdks/python/README.md]: Hello world is used a sub-title in other examples as well. Renaming that to basic pipeline is fine, however that should happen everywhere. You can either undo this change or updates all hello worlds accordingly in the other examples.

**Comment by aaltay** [comment]
Thank you @mariapython

Could you update the description of the PR to explain what you are doing? For context to others, this is part of a series of changes to fix the documentation to use the correct name of the SDK and also update examples to match current state. I also recommend creating a Jira issue for doing and tracking progress there.
 
Could you also add a note to the very top of the README.md about this ongoing change. Until this file is fully updated, it will have examples both in old and new form. This would be confusing for the users.

**Comment by mariapython** [comment]
Thanks, @aaltay.

I incorporated your comments and added a few more commits.

File: sdks/python/README.md [hunk 3]
```diff
@@ -16,6 +16,7 @@
     specific language governing permissions and limitations
     under the License.
 -->
+== This page is currently being updated. ==
```
Comment (aaltay) [inline sdks/python/README.md]: Did you mean `==` to render as something (like bold, header etc)?

Could you explain in a sentence what is being updated or link to the issue.

File: sdks/python/README.md [hunk 4]
```diff
@@ -257,78 +258,77 @@ Some interesting classes to navigate to:
...
- | df.FlatMap('add greetings',
-              lambda name, msgs: ['%s %s!' % (m, name) for m in msgs],
-              ['Hello', 'Hola'])
- | df.Write('save', df.io.TextFileSink('./greetings')))
+ | 'load messages' >> beam.Read(beam.io.TextFileSource('./names'))
+ | 'add greetings' >> beam.FlatMap(
+        lambda name, msgs: ['%s %s!' % (m, name) for m in msgs],
+        ['Hello', 'Hola']
+    )
+ | beam.Write('save', beam.io.TextFileSink('./greetings')))
```
Comment (aaltay) [inline sdks/python/README.md]: beam.Write() needs to be converted to the new format too.

File: sdks/python/README.md [hunk 5]
```diff
@@ -257,78 +258,77 @@ Some interesting classes to navigate to:
...
 The callable of a `FlatMap` can be a generator, that is,
 a function using `yield`.
 
 ```python
-import google.cloud.dataflow as df
-p = df.Pipeline('DirectPipelineRunner')
-# Add greetings using a FlatMap function using yield.
+import apache_beam as beam
+p = beam.Pipeline('DirectPipelineRunner')
+# Read a file containing names, add two greetings to each name (using yield), and write to a file.
```
Comment (aaltay) [inline sdks/python/README.md]: I believe having `using a FlatMap function using yield` in the comment was informative. Also please break long lines into two.

File: sdks/python/README.md (hunk 3 above)
Comment (mariapython) [inline sdks/python/README.md]: == was not meant to render, I just liked the view.
In the next commit, I'll explain that all the examples have been updated, but other references to alpha have not. Cyril just told me that he filed a bug for Eric to take care of that.

File: sdks/python/README.md [hunk 6]
```diff
@@ -255,80 +256,80 @@ Some interesting classes to navigate to:
...
 
-## Some Simple Examples
+## Simple Examples
 
-### Hello world
+### Basic pipeline
 
-Create a transform from an iterable and use the pipe operator to chain
-transforms:
+A basic pipeline creates a PCollection from an iterable
```
Comment (aaltay) [inline sdks/python/README.md]: Maybe "pipeline *that* creates"?

File: sdks/python/README.md [hunk 7]
```diff
@@ -339,89 +340,91 @@ text file from [Google Cloud Storage](https://cloud.google.com/storage/).
...
 (p
- | df.Read('read',
-           df.io.TextFileSource(
-           'gs://dataflow-samples/shakespeare/kinglear.txt'))
- | df.FlatMap('split', lambda x: re.findall(r'\w+', x))
- | df.combiners.Count.PerElement('count words')
- | df.Write('write', df.io.TextFileSink('./results')))
+ | 'read' >> beam.Read(
+        beam.io.TextFileSource('gs://dataflow-samples/shakespeare/kinglear.txt')
+    )
```
Comment (aaltay) [inline sdks/python/README.md]: I believe there is an extra space before )

File: sdks/python/README.md [hunk 8]
```diff
@@ -339,89 +340,91 @@ text file from [Google Cloud Storage](https://cloud.google.com/storage/).
...
 
-Here we use `GroupByKey` to count the words.
-This is a somewhat forced example of `GroupByKey`; normally one would use
-the transform `df.combiners.Count.PerElement`, as in the previous example.
-The example also shows the use of a wild-card in specifying the text file
-source.
-
+This is a somewhat forced example of `GroupByKey` to count the words.
+Normally, one would use the transform `beam.combiners.Count.PerElement`,
+as in the previous example. A wildcard is used to specify the text
```
Comment (aaltay) [inline sdks/python/README.md]: "The example also shows the use of a wildcard" part adds meaning here I think.

File: sdks/python/README.md [hunk 9]
```diff
@@ -339,89 +340,91 @@ text file from [Google Cloud Storage](https://cloud.google.com/storage/).
...
 
 ```python
-import google.cloud.dataflow as df
+import apache_beam as beam
+project = 'DESTINATION-PROJECT-ID'
 input_table = 'clouddataflow-readonly:samples.weather_stations'
-project = 'YOUR-PROJECT'
-output_table = 'DATASET.TABLENAME'
-p = df.Pipeline(argv=['--project', project])
+output_table = 'DESTINATION-DATASET.DESTINATION-TABLE'
```
Comment (aaltay) [inline sdks/python/README.md]: Should it be DESTINATION-DATASET:DESTINATION-TABLE with a : separator.

File: sdks/python/README.md [hunk 10]
```diff
@@ -430,40 +433,40 @@ number of tornadoes per month, but uses a query to filter out input instead
...
 (p
-| df.Read('read', df.io.BigQuerySource(query=input_query))
-| df.Write('write', df.io.BigQuerySink(
-    output_table,
-    schema='month:INTEGER, tornado_count:INTEGER',
-    create_disposition=df.io.BigQueryDisposition.CREATE_IF_NEEDED,
-    write_disposition=df.io.BigQueryDisposition.WRITE_TRUNCATE)))
+| 'read' >> beam.Read(beam.io.BigQuerySource(query=input_query))
+| 'save' >> beam.Write(beam.io.BigQuerySink(
+        output_table,
```
Comment (aaltay) [inline sdks/python/README.md]: Why there is an indentation change here?

File: sdks/python/README.md (hunk 8 above)
Comment (mariapython) [inline sdks/python/README.md]: I have kept that, not sure what you mean.

File: sdks/python/README.md (hunk 9 above)
Comment (mariapython) [inline sdks/python/README.md]: No. This is how I ran my example:
project = 'google.com:clouddfe'
output_table = 'beam_examples.output_tables_beam_examples'
input_query = 'SELECT month, COUNT(month) AS tornado_count ' \
        'FROM [clouddataflow-readonly:samples.weather_stations] ' \
        'WHERE tornado=true GROUP BY month'

File: sdks/python/README.md (hunk 7 above)
Comment (mariapython) [inline sdks/python/README.md]: I think you mean I used 4 space tab instead of 2, so I'll move that

Comment (aaltay) [inline sdks/python/README.md]: I meant that the `|` and `)` does not look aligned. I might be seeing wrong.

Also you are right that 2 spaces is the current standard for this code base.

File: sdks/python/README.md (hunk 9 above)
Comment (aaltay) [inline sdks/python/README.md]: Sounds good. thanks for checking.

File: sdks/python/README.md (hunk 8 above)
Comment (aaltay) [inline sdks/python/README.md]: This is fine either way.

I mean keeping the whole sentence "The example also shows the use of a wild-card in specifying the text file source." would be better than just "A wildcard is used to specify the text file source"

File: sdks/python/README.md [hunk 11]
```diff
@@ -255,215 +256,212 @@ Some interesting classes to navigate to:
...
+import apache_beam as beam
+from apache_beam.typehints import typehints
+p = beam.Pipeline('DirectPipelineRunner')
 (p
- | df.Read('A', df.io.TextFileSource('./names'))
- | df.Map('B1', lambda x: (x, 1)).with_output_types(typehints.KV[str, int])
- | df.GroupByKey('GBK')
- | df.Write('C', df.io.TextFileSink('./results')))
+ | 'read' >> beam.Read(beam.io.TextFileSource('./names'))
+ | 'add types' >> beam.Map(lambda x: (x, 1)).with_output_types(typehints.KV[unicode, int])
```
Comment (aaltay) [inline sdks/python/README.md]: What is reason for the change from `str` to `unicode` here?

File: sdks/python/README.md [hunk 12]
```diff
@@ -255,215 +256,212 @@ Some interesting classes to navigate to:
...
 
-## Some Simple Examples
+## Simple Examples
 
-### Hello world
+### Basic pipeline
 
-Create a transform from an iterable and use the pipe operator to chain
-transforms:
+A basic pipeline creates a `PCollection` from an iterable
```
Comment (aaltay) [inline sdks/python/README.md]: I think this needs to be somehow reworded.

Pipeline does not create a pcollection. Create transform create a pcollection from an iterable. It is a pipeline where that create transform is chained to other transforms with a pipe operator.

(You may leave changing this to another PR)

File: sdks/python/README.md (hunk 11 above)
Comment (mariapython) [inline sdks/python/README.md]: Great catch! I must say I have no explanation. I don't remember changing it (that is why I didn't look for it in the diff), not sure what happened. Just removed it

**Comment by aaltay** [comment]
LGTM. Thank you @mariapython

R: @dhalperi 

Dan, this is part of the effort to update the documentation. This PR changes the examples to use the correct APIs. As a follow up to this there will be rewrite of the remaining parts of this README (@hadarhod87) to correct names and references.

**Comment by dhalperi** [comment]
LGTM, merged. Close PR at your leisure.

Btw, when I rebase & squash (see the contribution guide) the commits we need a good way to set the commit message. One option (preferred, I think) is to use the first commit message. Since the first commit message was not especially informative, I wrote a new one based on the PR title.

(Does that make sense?)

**Comment by aaltay** [comment]
Thank you @dhalperi 

My opinion is that the first commit message or the PR message should have a very good description of the change and could be used as the summary commit message after squash. Follow up commits are usually fixups and they don't usually change the nature of the PR. Of course there are exceptions to the rule.

It would be useful if we can update the contribution guide (or maybe more practically the PR template) to encourage everyone to have a very informative PR message.

**Comment by dhalperi** [comment]
- [ ] test
- [ ] test2
